assembly/src/parsers/u32_ops.rs

```diff
@@ -29,175 +29,123 @@ pub fn parse_u32assert(op: &Token) -> Result<Node, ParsingError> {
     }
 }
 
-/// Returns `U32CheckedAdd` instruction node if no immediate value is provided or
-/// `U32CheckedAddImm` instruction node otherwise.
-///
-/// # Errors
-/// Returns an error if the instruction token contains wrong number of parameters, or if the
-/// provided parameter is not a u32 value.
-pub fn parse_u32checked_add(op: &Token) -> Result<Node, ParsingError> {
-    debug_assert_eq!(op.parts()[0], "u32checked_add");
-    match op.num_parts() {
-        0 => unreachable!(),
-        1 => Ok(Instruction(U32CheckedAdd)),
-        2 => {
-            let value = parse_param::<u32>(op, 1)?;
-            Ok(Instruction(U32CheckedAddImm(value)))
+macro_rules! parseop {
+    ($(#[$attr:meta])* fn $fn: ident $op: literal => ( $ins: ident , $imm_ins: ident)) => {
+        $(#[$attr])*
+        pub fn $fn(op: &Token) -> Result<Node, ParsingError> {
+            debug_assert_eq!(op.parts()[0], $op);
+            match op.num_parts() {
+                0 => unreachable!(),
+                1 => Ok(Instruction($ins)),
+                2 => {
+                    let value = parse_param::<u32>(op, 1)?;
+                    Ok(Instruction($imm_ins(value)))
+                }
+                _ => Err(ParsingError::extra_param(op)),
+            }
         }
-        _ => Err(ParsingError::extra_param(op)),
-    }
+    };
 }
 
-/// Returns `U32WrappingAdd` instruction node if no immediate value is provided or
-/// `U32WrappingAddImm` instruction node otherwise.
-///
-/// # Errors
-/// Returns an error if the instruction token contains wrong number of parameters, or if the
-/// provided parameter is not a u32 value.
-pub fn parse_u32wrapping_add(op: &Token) -> Result<Node, ParsingError> {
-    debug_assert_eq!(op.parts()[0], "u32wrapping_add");
-    match op.num_parts() {
-        0 => unreachable!(),
-        1 => Ok(Instruction(U32WrappingAdd)),
-        2 => {
-            let value = parse_param::<u32>(op, 1)?;
-            Ok(Instruction(U32WrappingAddImm(value)))
-        }
-        _ => Err(ParsingError::extra_param(op)),
-    }
-}
+parseop!(
+    /// Returns `U32CheckedAdd` instruction node if no immediate value is provided or
+    /// `U32CheckedAddImm` instruction node otherwise.
+    ///
+    /// # Errors
+    /// Returns an error if the instruction token contains wrong number of parameters, or if the
+    /// provided parameter is not a u32 value.
+    fn parse_u32checked_add "u32checked_add" => (U32CheckedAdd, U32CheckedAddImm)
```

Comment:
I experimented a bit with other syntax. This is an option that is concise and readable, but better suggestions are welcome :)